### PR TITLE
feat(insights): conversion cohorts (5.1/5.2) + weekly pre-warm

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guidelines](https://github.com/Automattic/newspack-workspace/blob/main/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Wires the **Conversion Journey** tab's Section 5 cohort-retention charts (5.1 and 5.2) to live WP/Woo data, replacing the `coming_soon` placeholders, and serves them from a weekly pre-warmed snapshot so the expensive queries never run on a reader request.

- **5.1 Registration → Conversion cohort** — for each monthly cohort of registered readers (by `user_registered`), the cumulative share that converted to subscriber **or** donor by month N. Reference line: 15% at 6 months.
- **5.2 Subscriber Retention cohort** — for each monthly cohort of new subscribers (by first non-donation subscription), the share still active at month N. Both cancellations and natural expirations count as churn. Reference line: 70% at 12 months.
- **Weekly pre-warm** — a recurring Action Scheduler job (Monday 06:00 UTC) computes both cohorts and stores them in a new, reusable window-independent snapshot cache tier. Reader requests read the snapshot; on a cold cache they degrade gracefully to the existing "coming soon" state and schedule a one-off background refresh, so cohorts never block or slow a page load.

Both cohorts are window-independent snapshots, so the React layer and the tab's response contract are unchanged. The demo/fixture payloads intentionally remain "coming soon" (live demo curves are out of scope here).

Part of NPPD-1692 (Conversion Journey Phase B); targets the `feat/insights-rsm` epic branch. (The original cohort-pre-warm ticket NPPD-1606 was canceled; its per-publisher pre-warm objection was specific to the BigQuery cache, whereas these cohorts are pure WP/Woo, so a local weekly pre-warm is the right layer.)

### How to test the changes in this Pull Request:

> ⚠️ **Live verification is the mandatory gate.** The WooCommerce SQL (reader population, subscription cohort intervals, HPOS + legacy) cannot be exercised by the PHPUnit harness (no WC tables), so it must be checked on a real Audience site. The unit suite proves the cache tier, the cohort math, and the request-path/snapshot wiring only.

1. **Unit tests** — from `plugins/newspack-plugin`: `n test-php --group insights` (expect all green) and `n test-php --filter Newspack_Test_Insights_Cache` (snapshot tier). PHPCS clean on the host.
2. **Live site (e.g. blockclub)** — load the Insights → Conversion Journey tab. On first load the two Section 5 charts may show "coming soon" while the one-off refresh runs; reload after the background job completes and confirm both cohort charts render with data.
3. **5.1** — confirm the reader population and per-cohort cumulative conversion look plausible, the oldest cohort reaches ~12 months, and the 15%-at-6-months reference line shows.
4. **5.2** — spot-check a known **cancelled** subscription and a known **expired** (`_schedule_end`) subscription: each should drop out of the active count at the correct month offset. Confirm the 70%-at-12-months reference line.
5. **HPOS + legacy parity** — verify on both a HPOS site and a legacy-CPT-order site that the cohorts populate equivalently.
6. **Pre-warm** — trigger `run_cohort_refresh` (WP-CLI / Action Scheduler admin) and confirm the snapshot transient `newspack_insights_conversion_<md5(["cohorts"])>` populates and the request path then serves it warm. Confirm the **weekly recurring** action is scheduled: `as_next_scheduled_action( 'newspack_insights_conversion_cohort_refresh_weekly', [], 'newspack-insights' )` returns a timestamp.
7. **Graceful degradation** — with the snapshot cold/cleared, confirm the tab still renders (cohorts show "coming soon") and does not error or hang.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: plugins/newspack-plugin/includes/wizards/insights/storage/class-hpos-storage.php [hunk 1]
```diff
@@ -1507,4 +1509,76 @@ private function shape_attributed_subscription_rows( array $rows ): array {
...
+					JOIN {$prefix}wc_orders_meta sm2
+						ON sm2.order_id = o2.id AND sm2.meta_key = '_schedule_start'
+					JOIN {$prefix}woocommerce_order_items oi2
+						ON oi2.order_id = o2.id AND oi2.order_item_type = 'line_item'
+					JOIN {$prefix}woocommerce_order_itemmeta oim2
+						ON oim2.order_item_id = oi2.order_item_id AND oim2.meta_key = '_product_id'
+					WHERE o2.type = 'shop_subscription'
+					  AND oim2.meta_value NOT IN ($donations)
+					  AND sm2.meta_value != ''
+					GROUP BY o2.customer_id
```
[the remark this answers is not in the document]
Comment (dkoo) [reply plugins/newspack-plugin/includes/wizards/insights/storage/class-hpos-storage.php:1569]: Good catch — added `AND o.customer_id > 0` to both the outer query and the inner cohort-defining subquery, mirroring `get_new_subscriber_records_in_window()`. Fixed in f1e8e8b01.

File: plugins/newspack-plugin/includes/wizards/insights/storage/class-legacy-storage.php [hunk 2]
```diff
@@ -1436,4 +1438,75 @@ private function shape_attributed_subscription_rows( array $rows ): array {
...
+					JOIN {$prefix}postmeta sm2
+						ON sm2.post_id = p2.ID AND sm2.meta_key = '_schedule_start'
+					JOIN {$prefix}woocommerce_order_items oi2
+						ON oi2.order_id = p2.ID AND oi2.order_item_type = 'line_item'
+					JOIN {$prefix}woocommerce_order_itemmeta oim2
+						ON oim2.order_item_id = oi2.order_item_id AND oim2.meta_key = '_product_id'
+					WHERE p2.post_type = 'shop_subscription'
+					  AND oim2.meta_value NOT IN ($donations)
+					  AND sm2.meta_value != ''
+					GROUP BY CAST(cust2.meta_value AS UNSIGNED)
```
[the remark this answers is not in the document]
Comment (dkoo) [reply plugins/newspack-plugin/includes/wizards/insights/storage/class-legacy-storage.php:1497]: Fixed in f1e8e8b01 — added `CAST(cust.meta_value AS UNSIGNED) > 0` to both the outer query and the inner cohort subquery, matching the legacy sibling queries.

File: plugins/newspack-plugin/includes/wizards/insights/metrics/class-conversion-metric.php [hunk 3]
```diff
@@ -1749,4 +1906,192 @@ public function get_top_pages_no_conversion( DateTimeInterface $start, DateTimeI
...
+	 */
+	public static function maybe_schedule_cohort_prewarm(): void {
+		if ( ! function_exists( 'as_schedule_recurring_action' ) || ! function_exists( 'as_next_scheduled_action' ) ) {
+			return;
+		}
+		if ( false !== as_next_scheduled_action( self::COHORT_REFRESH_WEEKLY_ACTION, [], self::COHORT_REFRESH_GROUP ) ) {
+			return;
+		}
+		$next = ( new \DateTimeImmutable( 'next monday 06:00', new \DateTimeZone( 'UTC' ) ) )->getTimestamp();
+		as_schedule_recurring_action( $next, WEEK_IN_SECONDS, self::COHORT_REFRESH_WEEKLY_ACTION, [], self::COHORT_REFRESH_GROUP );
```
[the remark this answers is not in the document]
Comment (dkoo) [reply plugins/newspack-plugin/includes/wizards/insights/metrics/class-conversion-metric.php]: Fixed in f1e8e8b01 — extracted `next_weekly_prewarm_timestamp()` that computes this week's Monday 06:00 UTC and only rolls forward a week if that instant has already passed, so an early-Monday activation no longer skips the upcoming run. Added a unit test covering the Monday-05:00 edge case.

File: plugins/newspack-plugin/includes/wizards/insights/class-cache.php [hunk 4]
```diff
@@ -110,6 +116,26 @@ private static function read_cached( string $tab, array $key_parts ): ?array {
...
+	 * @param string[] $key_parts Canonicalized key components.
+	 * @return array{ payload: array, computed_at: string, source: string }|null
+	 */
+	public static function peek( string $tab, string $source, array $key_parts ): ?array {
+		unset( $source ); // Reserved for parity with store()/refresh(); key is tab+parts.
+		if ( self::is_disabled() ) {
+			return null;
+		}
+		return self::read_cached( $tab, $key_parts );
+	}
```
[the remark this answers is not in the document]
Comment (dkoo) [reply plugins/newspack-plugin/includes/wizards/insights/class-cache.php:141]: Fixed in f1e8e8b01 — `peek()` now validates the cached envelope's source and returns null on mismatch. Added a test asserting a SOURCE_SNAPSHOT write is not returned for a SOURCE_EXTERNAL peek.
